runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -378,6 +378,8 @@ struct AMDAIEDeviceModel {
   DenseMap<uint32_t, SmallVector<uint32_t>> getChannelToValidBdIds(
       AMDAIETileType tileType) const;
 
+  uint8_t getDmaMaxQueueSize(uint8_t col, uint8_t row);
```

Comment:
Really unfortunate that this can't be `const`.  I see that getTileType uses a const_cast to workaround the lack of const-correctness in aie_rt, seems like first done here 

https://github.com/nod-ai/iree-amd-aie/commit/6c4f9050d8b9166845d95dc0a404e47d24938fd3#diff-17008229092a63d5df9831105108aa99c799a226441b0dd9d8327708e57fc2aeR218

I tracked this down because I saw you passing `AMDAIE::AMDAIEDeviceModel deviceModel` by value, which isn't great as this isn't a pointer type so there is overhead. But if getDmaMaxQueueSize isn't const, deviceModel can't be const ref where you use it...